KBaseSearchEngine.spec

```diff
@@ -261,6 +261,8 @@ module KBaseSearchEngine {
           pagination and sorting.
       total - total number of found objects.
       search_time - common time in milliseconds spent.
+      list<GUID> removedGuids - list of result GUIDs that are inaccessible or deleted in the workspace but have
+          not been updated in search. Returned when setting env SHOWREMOVEDGUIDS to anything. Returned in additional properties
```

Comment:
Also, we should clearly document the behavior of the search methods and that they may not always return $limit objects, even when there are more than $limit objects to go.